From 9c7072d18b744a4513716e23e7908ecb7ad1cb3f Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Mon, 6 Feb 2023 10:14:22 +0800
Subject: [PATCH 17/45] Revert "drm/ingenic: Fix missing
 platform_driver_unregister() call in ingenic_drm_init()"

This reverts commit 68af1a4842b4e69e1ac4a3488fad387972e98ee0.
---
 drivers/gpu/drm/ingenic/ingenic-drm-drv.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/drivers/gpu/drm/ingenic/ingenic-drm-drv.c b/drivers/gpu/drm/ingenic/ingenic-drm-drv.c
index 5ec9770e401e4..e34718cf5c2e9 100644
--- a/drivers/gpu/drm/ingenic/ingenic-drm-drv.c
+++ b/drivers/gpu/drm/ingenic/ingenic-drm-drv.c
@@ -1120,11 +1120,7 @@ static int ingenic_drm_init(void)
 			return err;
 	}
 
-	err = platform_driver_register(&ingenic_drm_driver);
-	if (IS_ENABLED(CONFIG_DRM_INGENIC_IPU) && err)
-		platform_driver_unregister(ingenic_ipu_driver_ptr);
-
-	return err;
+	return platform_driver_register(&ingenic_drm_driver);
 }
 module_init(ingenic_drm_init);
 
-- 
2.25.1

